### ISSUE TITLE
quotes: avoidEscape option allow the usage of template string

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#10627]
**Tell us about your environment**

* **ESLint Version:** 5.1.0/master
* **Node Version:** n/w
* **npm Version:** n/w

**What parser (default, Babel-ESLint, etc.) are you using?**default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
extends: ['eslint'],
  rules: {
    "quotes": ["error", "double", {"avoidEscape": true}]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var s = `"hello world"`;
```

**What did you expect to happen?**
no error reported.
**What actually happened? Please include the actual, raw output from ESLint.**

![image](https://user-images.githubusercontent.com/13050025/42949584-9f9daa8a-8ba4-11e8-93b4-1ef3f2be6d61.png)

**Comment by ilyavolodin** [comment]
Doesn't look like an issue to me. You specified that you want double quotes, but then you use template quotes for a string. Seems like it's working as designed.

**Comment by Aladdin-ADD** [comment]
that was true, but I have enabled `avoidescape` option ( https://eslint.org/docs/rules/quotes#avoidescape )

**Comment by ilyavolodin** [comment]
If I remember it correctly, avoidescape only works for single and double quotes. I think it was never implemented for templates.

**Comment by mdjermanovic** [comment]
Issue #5234 - request to allow backticks as a way to avoid escaped single quotes.

In the thread there was a [proposal](https://github.com/eslint/eslint/issues/5234#issuecomment-183698009) which is, I think, same as this one, but it was dismissed (the first two replies below the proposal).

The problem is that the same rule also [disallows templates that can be replaced with strings](https://github.com/eslint/eslint/issues/3090) and it seems that this has precedence over `avoidEscape` (#5147).

#5234 was resolved by adding `allowTemplateLiterals` which simply ignores all template literals (this might be a bug, but that's another subject).

So, after reading all these threads, I believe that the current behavior is correct, i.e. this should be reported as an error.

**Comment by kaicataldo** [comment]
@aladdin-add Are you still championing this?

**Comment by FanFataL** [comment]
Bump

**Comment by kaicataldo** [comment]
@aladdin-add One more friendly ping. We should probably close this if this is no longer being championed.

**Comment by aladdin-add** [comment]
ops, thanks for the ping! I like the idea in https://github.com/eslint/eslint/issues/12156, will champion it.
